osf/models/analytics.py

```diff
@@ -99,34 +96,24 @@ def clean_page(page):
             '$', '_'
         )
 
-    @staticmethod
-    def deconstruct_id(page):
-        """
-        Backwards compatible code for use in writing to both _id field and
-        action, resource, file, and version fields simultaneously.
-        """
-        split = page.split(':')
-        action = split[0]
-        resource = Guid.load(split[1])
-        file = BaseFileNode.load(split[2])
-        if len(split) == 3:
-            version = None
+    @classmethod
+    def update_counter(cls, resource, file, version, action, node_info):
+        if version is not None:
+            page = '{0}:{1}:{2}:{3}'.format(action, resource._id, file._id, version)
         else:
-            version = split[3]
-        return resource, file, action, version
+            page = '{0}:{1}:{2}'.format(action, resource._id, file._id)
 
-    @classmethod
-    def update_counter(cls, page, node_info):
         cleaned_page = cls.clean_page(page)
         date = timezone.now()
         date_string = date.strftime('%Y/%m/%d')
         visited_by_date = session.data.get('visited_by_date', {'date': date_string, 'pages': []})
         with transaction.atomic():
-            # Temporary backwards compat - when creating new PageCounters, temporarily write to _id, resource, file, action, and version fields
+            # Temporary backwards compat - when creating new PageCounters, temporarily keep writing to
+            # After we're sure this is stable, we can stop writing to the _id field, and query on
+            # resource/file/action/version
             try:
                 model_instance = cls.objects.get(_id=cleaned_page)
```

Comment:
Temporarily still looking up the page_counter on _id.  We're now writing to five fields on creation - _id, resoure, file, action, and version.  

After this has been out awhile and it's stable, then we can remove the _id field, using a long id or similar instead, and migrating all the old _ids in the db.  After that, we can start querying on resource/file/action/version combo.